### PR TITLE
Bugfix Plugin deployment

### PR DESCRIPTION
**Pull request by Klaas-Ritense**
- Plugins were not always deployed. This broke the IntegrationTests. Cause: TransactionalEventListener only runs when inside a Transaction.

- Stackoverflow due to ManyToOne relation. Fix: first remove all plugin-properties/action/action-properties. Because these have a ManyToOne relation with ProcessDefinition.

- Occasional error in IntelliJ: `Expiring Daemon because JVM heap space is exhausted`. FIx: additional jvmargs.

### CODE REVIEW

File: plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt [hunk 1]
```diff
@@ -29,23 +28,30 @@ import com.ritense.plugin.exception.PluginDefinitionNotDeployedException
...
+    private val pluginPropertyRepository: PluginPropertyRepository,
     private val pluginActionDefinitionRepository: PluginActionDefinitionRepository,
     private val pluginActionPropertyDefinitionRepository: PluginActionPropertyDefinitionRepository
 ) {
 
-    @TransactionalEventListener(ApplicationStartedEvent::class)
+    @EventListener(ApplicationStartedEvent::class)
     fun deployPluginDefinitions() {
         logger.info { "Deploying plugins" }
+        pluginPropertyRepository.deleteAll()
```
Comment (marijnritense) [inline plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt:51]: This causes a lot of unnecessary deletions, maybe change it so deletions only happen if there were changes?

File: plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt [hunk 2]
```diff
@@ -29,23 +28,30 @@ import com.ritense.plugin.exception.PluginDefinitionNotDeployedException
...
 class PluginDeploymentListener(
     private val pluginDefinitionResolver: PluginDefinitionResolver,
     private val pluginDefinitionRepository: PluginDefinitionRepository,
+    private val pluginPropertyRepository: PluginPropertyRepository,
     private val pluginActionDefinitionRepository: PluginActionDefinitionRepository,
     private val pluginActionPropertyDefinitionRepository: PluginActionPropertyDefinitionRepository
 ) {
 
-    @TransactionalEventListener(ApplicationStartedEvent::class)
+    @EventListener(ApplicationStartedEvent::class)
```
Comment (marijnritense) [inline plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt:48]: Wouldn't adding the transactional annotation to the test also work? That allows us to keep the annotation as-is here.

File: plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt:51]: unfortunately, that won't fix the stackoverflow error. This error happens even when nothing changed at all: On every second application start-up the and the plugins are deployed for the second time, the error happens

File: plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt (hunk 2 above)
Comment (Klaas-Ritense) [reply plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt:48]: After starting the application locally, I found that plugins were not deployed at all anymore.

**Review by marijnritense — APPROVED** [review]

**Pull request merged**